tests/functional/base_test.go

```diff
@@ -212,11 +240,15 @@ func OVNControllerConditionGetter(name types.NamespacedName) condition.Condition
 }
 
 func SetExternalEndpoint(name types.NamespacedName, endpoint string) {
-	Eventually(func(g Gomega) {
-		cluster := GetOVNDBCluster(name)
-		cluster.Status.DBAddress = endpoint
-		g.Expect(k8sClient.Status().Update(ctx, cluster)).To(Succeed())
-	}, timeout, interval).Should(Succeed())
+	if endpoint != "" {
+		Eventually(func(g Gomega) {
+			cluster := GetOVNDBCluster(name)
+			cluster.Status.DBAddress = endpoint
+			g.Expect(k8sClient.Status().Update(ctx, cluster)).To(Succeed())
+		}, timeout, interval).Should(Succeed())
+	} else {
+		ScaleDBCluster(name, 0)
```

Comment:
This function is called twice with endpoint = "":
- [should delete the external config map when networkAttachment is detached from SB DB](https://github.com/averdagu/ovn-operator/blob/feat/sbdb-dns/tests/functional/ovncontroller_controller_test.go#L225)
- [should delete an external ConfigMap once SB DBCluster is detached from NAD](https://github.com/averdagu/ovn-operator/blob/feat/sbdb-dns/tests/functional/ovncontroller_controller_test.go#L510)

The first one (the one you refer on the comment) works fine without ScaleDBCluster = 0.So updating NAD to == is enough.

The problem is with second test, which sets ExternalEndpoint to "" (without removing NAD) and expects that CM does not exist. The problem is that since even if they set Status.dbAddress = "", if pods still exist, and pods have NAD != "", whenever a reconcile loop is triggered, status.dbAddress will be the DNS name.

Given that in order to set the Status.dbAddress is to scale down to 0 or remove the NAD (as the other test does) at the time where I was developing the tests I did the first option.

So in order to keep consistency bw both tests, I'm modifying the test to actually remove the NAD from cluster.

Reply:
Resolving conversation as this function has been removed